tests/test_pages.py

```diff
@@ -60,22 +61,39 @@ def chrome_options(chrome_options):
     return chrome_options
 
 
-@pytest.fixture
-def user(variables):
+def _user(variables):
     return User(variables.get('username', 'admin'), variables.get('password', 'changeme'),
                 variables.get('name', 'Admin User'))
 
 
-@pytest.fixture(params=PAGES)
-def url(request, base_url):
-    if not base_url:
-        raise ValueError('Base URL not specified')
-    return base_url + request.param
+@pytest.fixture(scope='session')
+def user(variables):
+    return _user(variables)
+
+
+def pytest_generate_tests(metafunc):
+    if 'url' in metafunc.fixturenames:
+        base_url = metafunc.config.option.base_url
+        if not base_url:
+            assert base_url
+
+        # mimic the variables fixture from pytest-variables
+        user_obj = _user(metafunc.config._variables)  # pylint: disable=protected-access
+
+        response = requests.get(f'{base_url}/menu', auth=(user_obj.username, user_obj.password))
+        if response.status_code == 404:
+            # Menu is only available since Foreman 2.0
+            pages = [base_url + page for page in PAGES]
+        else:
+            assert response
```

Comment:
what is it asserting here?

Reply:
`bool(response)` is equal to `response.ok`
https://github.com/psf/requests/blob/eedd67462819f8dbf8c1c32e77f9070606605231/requests/models.py#L673-L681

Reply:
That's, uh, "interesting". I'm used to raise for status or similar, but yeah, that works too.